src/core/network/sendBeaconFactory.js

```diff
@@ -17,6 +17,13 @@ export default (navigator, fetch, logger) => {
       logger.log("The `beacon` call has failed; falling back to `fetch`");
       return fetch(url, body);
     }
-    return Promise.resolve();
+
+    // Using sendBeacon, we technically don't get a response back from
+    // the server, but we'll resolve the promise with an object to maintain
+    // consistency with other network strategies.
+    return Promise.resolve({
+      status: 204,
+      body: ""
+    });
```

Comment:
I didn't need to do this but it made code and tests simpler and more consistent.